dbt/node_runners.py

```diff
@@ -291,11 +291,13 @@ def run_hooks(cls, project, adapter, flat_graph, hook_type):
             model_name = compiled.get('name')
             statement = compiled['wrapped_sql']
 
-            hook_dict = dbt.hooks.get_hook_dict(statement)
+            hook_index = hook.get('index', len(hooks))
+            hook_dict = dbt.hooks.get_hook_dict(statement, index=hook_index)
             compiled_hooks.append(hook_dict)
 
-        for hook in compiled_hooks:
+        ordered_hooks = sorted(compiled_hooks, key=lambda h: h.get('index', 0))
```

Comment:
@cmcarthur the actual fix is that we sort by this `index` here, and run the hooks in sorted order

Reply:
i get that, but when would the index ordering here be different than the ordering of hooks